Fix writing 64 bits integer into PLY files

MaelRL opened the pull request:
## Summary of Changes

The PLY file format does not support 64 bits (signed/unsigned) integers.

In the overload of the writer specific to `Surface_mesh`, we try to write all property maps with value types compatible with the PLY file format (char, uchar, etc.).

We also tolerate 64 bits signed / unsigned integers, but used to cast it to int32 / uint32:

```cpp
    {
      Int64_map pmap;
      boost::tie(pmap, okay) = sm.template property_map<Simplex,boost::int64_t>(prop[i]);
      if(okay)
      {
        os << "property int " << name << std::endl;
        printers.push_back(new internal::Simple_property_printer<Simplex,Int64_map,boost::int32_t>(pmap));
        continue;
      }
    }
```

In https://github.com/CGAL/cgal/pull/6575, the code was factorized, but the conversion to 32 bits was accidentally lost (https://github.com/CGAL/cgal/commit/aa9f5215c4afa79d29cc5491d1d690cd2ed7a142). Hence, we were writing 8 bytes values (while announcing 4 bytes values) and writing a property map with value type a 64 bits integers (or a range of 64 bits integers) would make the file unreadable.

The PR re-introduces the conversions, and also removes the recursion of the function.

The bug is from CGAL 5.5, but we have recent changes to PLY I/O introducing named parameters and whatnot that make it a little tedious to backport so I will wait to see if there is a need.

## Release Management

* Affected package(s): `Surface_mesh`
* Issue(s) solved (if any): -
* Feature/Small Feature (if any): -
* License and copyright ownership: no change

MaelRL committed:
Remove types which cannot be read from a PLY file

MaelRL committed:
Fix writing 64 bits integers into PLY files + fold expressions

Broken by https://github.com/CGAL/cgal/pull/6575, specifically
the commit aa9f521.

MaelRL committed:
Add a test for PLY I/O of 64 bits integers

MaelRL committed:
Add a warning about Surface_mesh PLY writing for 64 bits integers

MaelRL committed:
Fix a warning

/mnt/testsuite/include/CGAL/IO/PLY/PLY_writer.h:296:30: warning:
loop variable 'v' of type 'const unsigned int&' binds to a temporary
constructed from type 'const long unsigned int' [-Wrange-loop-construct]
  296 |       for(const ElementType& v : vec)

This is because ElementType is not the value type of the vector,
but the data type that we write, and for 64 bits integer,
we convert to 32 bits. So in that case, we can't cast yet.

MaelRL committed:
Rewrite to C++17

afabri commented:
runtime error in [I-c-65](https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-6.2-Ic-65/Surface_mesh/TestReport_magritte_arm64-apple_bigsur_clang1200-release-64bits.gz)

MaelRL committed:
Debug helpers

MaelRL committed:
Explicit 'std::size_t' as an accepted value type of SM pmap for write_PLY

This needs to be made explicit because on macOS, std::size_t
is not the same as std::(u)int64_t

sloriot commented:
Successfully tested in CGAL-6.2-Ic-70
